# First hunk by Mikael Hallendal, fixing the crash in the ADD code.
# Second and third hunk by Martin Schlemmer, fixing the crash in the EDIT code.

--- filter/rule-editor.c.orig	Tue Dec  4 21:20:09 2001
+++ filter/rule-editor.c	Tue Apr  2 00:49:27 2002
@@ -253,7 +253,9 @@
 	default:
 		gnome_dialog_close (GNOME_DIALOG (dialog));
 	case -1:
+	  if (re->edit) {
 		gtk_object_unref (GTK_OBJECT (re->edit));
+	  }
 		re->edit = NULL;
 		
 		re->dialog = NULL;
@@ -327,7 +329,9 @@
 	default:
 		gnome_dialog_close (GNOME_DIALOG (dialog));
 	case -1:
+	  if (re->edit) {
 		gtk_object_unref (GTK_OBJECT (re->edit));
+	  }
 		re->edit = NULL;
 		
 		re->dialog = NULL;
@@ -361,7 +365,7 @@
 	
 	gtk_window_set_default_size (GTK_WINDOW (re->dialog), 600, 400);
 	gtk_window_set_policy (GTK_WINDOW (re->dialog), FALSE, TRUE, FALSE);
-	gtk_widget_set_parent (re->dialog, GTK_WIDGET (re));
+	gtk_widget_set_parent_window (re->dialog, GTK_WIDGET (re));
 	gtk_box_pack_start (GTK_BOX (GNOME_DIALOG (re->dialog)->vbox), rules, TRUE, TRUE, 0);
 	gtk_signal_connect (GTK_OBJECT (re->dialog), "clicked", edit_editor_clicked, re);
 	gtk_signal_connect (GTK_OBJECT (re->dialog), "destroy", edit_editor_destroyed, re);
